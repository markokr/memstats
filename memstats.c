
#include <postgres.h>
#include <fmgr.h>
#include <utils/memutils.h>

PG_MODULE_MAGIC;

Datum memstats(PG_FUNCTION_ARGS);
PG_FUNCTION_INFO_V1(memstats);

/*
 * launch MemoryContextStats(TopMemoryContext)
 */
Datum memstats(PG_FUNCTION_ARGS)
{
	static int counter;

	elog(NOTICE, "MEMSTATS %d", counter++);
	MemoryContextStats(TopMemoryContext);

	PG_RETURN_INT32(1);
}

