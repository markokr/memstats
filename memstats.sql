
CREATE OR REPLACE FUNCTION memstats() RETURNS int4 AS '$libdir/memstats' LANGUAGE C;

