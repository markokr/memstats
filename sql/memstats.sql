
\i memstats.sql

select memstats();

select memstats();

