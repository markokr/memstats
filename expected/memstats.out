\i memstats.sql
CREATE FUNCTION memstats() RETURNS int4 AS '$libdir/memstats' LANGUAGE C;
select memstats();
NOTICE:  MEMSTATS 0
 memstats 
----------
        1
(1 row)

select memstats();
NOTICE:  MEMSTATS 1
 memstats 
----------
        1
(1 row)

